check/src/cache.rs

```diff
@@ -34,89 +38,218 @@ sol! {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_builtin(&cfg.endpoint)
+        .await?;
+    let chain_id = provider.get_chain_id().await?;
+    let wallet = cfg.auth.alloy_wallet(chain_id)?;
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_builtin(&cfg.endpoint)
+        .await?;
+    let cache_manager_addr = get_cache_manager_address(provider.clone()).await?;
+    let cache_manager = CacheManager::new(cache_manager_addr, provider.clone());
+    let addr = cfg.address.to_fixed_bytes().into();
+    let mut place_bid_call = cache_manager.placeBid(addr).value(U256::from(cfg.bid));
+    if let Some(max_fee) = cfg.max_fee_per_gas_gwei {
+        place_bid_call = place_bid_call.max_fee_per_gas(gwei_to_wei(max_fee)?);
+    };
 
-    if verbose {
-        let gas = format_gas(receipt.gas_used.unwrap_or_default());
+    greyln!("Checking if contract can be cached...");
+
+    let raw_output = place_bid_call.clone().call().await;
+    if let Err(e) = raw_output {
+        let Error::TransportError(tperr) = e else {
+            bail!("failed to send cache bid tx: {:?}", e)
+        };
+        let Some(err_resp) = tperr.as_error_resp() else {
+            bail!("no error payload received in response: {:?}", tperr)
+        };
+        let Some(errs) = err_resp.as_decoded_error::<CacheManagerErrors>(true) else {
+            bail!("failed to decode CacheManager error: {:?}", err_resp)
+        };
+        handle_cache_manager_error(errs)?;
+    }
+    greyln!("Sending cache bid tx...");
+    let pending_tx = place_bid_call.send().await?;
+    let receipt = pending_tx.get_receipt().await?;
+    if cfg.verbose {
+        let gas = format_gas(receipt.gas_used);
         greyln!(
-            "Successfully cached contract at address: {address} {} {gas}",
+            "Successfully cached contract at address: {addr} {} {gas} gas used",
             "with".grey()
         );
     } else {
-        greyln!("Successfully cached contract at address: {address}");
+        greyln!("Successfully cached contract at address: {addr}");
     }
     let tx_hash = receipt.transaction_hash.debug_lavender();
-    greyln!("Sent Stylus cache tx with hash: {tx_hash}");
+    greyln!("Sent Stylus cache bid tx with hash: {tx_hash}");
     Ok(())
 }
+
+async fn get_cache_manager_address<P>(provider: P) -> Result<Address>
+where
+    P: Provider + Clone + Send + Sync,
+{
+    let arb_wasm_cache = ArbWasmCache::new(ARB_WASM_CACHE_ADDRESS, provider.clone());
+    let result = arb_wasm_cache.allCacheManagers().call().await?;
+    if result.managers.is_empty() {
+        bail!("no cache managers found in ArbWasmCache, perhaps the Stylus cache is not yet enabled on this chain");
+    }
+    Ok(*result.managers.last().unwrap())
+}
+
+fn format_gas(gas: u128) -> String {
+    let gas: u128 = gas.try_into().unwrap_or(u128::MAX);
```

Comment:
The linter pointed out that this line could be removed, and it made sense. Maybe in the future we should block PRs from being merged if we get linter warnings.